calyx-py/calyx/builder.py

```diff
@@ -131,7 +142,25 @@ def cell(
         self.index[name] = builder
         return builder
 
-    def reg(self, name: str, size: int):
+    def sub_component(
+        self,
+        cell_name: str,
+        sub_comp_name: str | ComponentBuilder,
```

Comment:
I know this is silly, but maybe this should just be called `comp_name` (and the docstring should just say "component")? I don't think we've used "sub-component" elsewhere, and I'm not sure what it means… we don't exactly have a hierarchy of components in Calyx (although we do have a hierarchy of cells).

Reply:
How do you feel about `comp_instance`?

Reply:
Good call!